src/app/services/schema/app-schema.service.ts

```diff
@@ -637,65 +639,74 @@ export class AppSchemaService {
       }));
   }
 
-  private handleSchemaSubQuestions(payload: CommonSchemaAddControl, newFormControl: CustomUntypedFormControl): void {
-    const {
-      schema, isNew, path, subscription, formGroup, config, isParentImmutable,
-    } = payload;
+  private handleSchemaSubQuestions({
+    schema,
+    isNew,
+    path,
+    subscription,
+    formGroup,
+    config,
+    isParentImmutable,
+  }: CommonSchemaAddControl, newFormControl: CustomUntypedFormControl): void {
+    if (!schema.subquestions) {
+      return;
+    }
 
-    if (schema.subquestions) {
-      schema.subquestions.forEach((subquestion) => {
-        subscription.add(
-          this.addFormControls({
-            isNew,
-            path,
-            chartSchemaNode: subquestion,
-            formGroup,
-            config,
-            isParentImmutable: !!schema.immutable || isParentImmutable,
-          }),
-        );
+    for (const subquestion of schema.subquestions) {
+      subscription.add(
+        this.addFormControls({
```

Comment:
This method name is not very intuitive. It should be named something that indicates what it returns. It's not just adding a form control. It's also returning a subscription to form control's value changes. Suggesstion: `getNewFormControlChangesSubscription` or maybe it can be divided between two methods. One to create a form control and one to get the subscription. Or, we can pass the subscription collection to the method so adding the subscription can be done inside the method. Something should be done.